### PR TITLE
Python: Add Support for CORS Middlewares

### PR DESCRIPTION
**Pull request by Kwstubbs**
Model FastAPI & Starlette CORS Middlewares for misconfiguration

### CODE REVIEW

**Review by tausbn — CHANGES_REQUESTED** [review]
Good stuff! 

I have made some comments and suggestions for improvement here and there. (Note: I have not commented on _everything_ that needs to be fixed -- for instance usage of `toString` or style guide violations -- so please exercise your judgement in applying these comments more broadly.)

Also, I guess the QHelp could do with a review from the docs team.

File: python/ql/lib/semmle/python/frameworks/FastApi.qll [hunk 1]
```diff
@@ -30,6 +30,51 @@ module FastApi {
...
+  class AddCorsMiddlewareCall extends Http::Server::CorsMiddleware::Range, AddMiddlewareCall {
+    /**
+     * Gets the string corresponding to the middleware
+     */
+    override string middleware_name() { result = this.getArg(0).asExpr().(Name).toString() }
+
+    /**
+     * Gets the dataflow node corresponding to the allowed CORS origins
+     */
+    override DataFlow::Node allowed_origins() { result = this.getArgByName("allow_origins") }
```
[the remark this answers is not in the document]
Comment (Kwstubbs) [reply python/ql/lib/semmle/python/frameworks/FastApi.qll:57]: They are passed by argument only.

File: python/ql/lib/semmle/python/frameworks/Starlette.qll [hunk 2]
```diff
@@ -25,6 +25,98 @@ private import semmle.python.frameworks.data.ModelsAsData
...
+    private DataFlow::TypeTrackingNode instance(DataFlow::TypeTracker t) {
+      t.start() and
+      result instanceof InstanceSource
+      or
+      exists(DataFlow::TypeTracker t2 | result = instance(t2).track(t2, t))
+    }
+
+    /** Gets a reference to an instance of `starlette.middleware.Middleware`. */
+    DataFlow::Node instance() { instance(DataFlow::TypeTracker::end()).flowsTo(result) }
+  }
```
[the remark this answers is not in the document]
Comment (Kwstubbs) [reply python/ql/lib/semmle/python/frameworks/Starlette.qll]: Yup, seems I overcooked here. 🆘  😄

**Review by tausbn — CHANGES_REQUESTED** [review]
Your changes look really good. Thanks!

Now there's just one thing missing: some tests for the concepts you're introducing.

For this, you'll need to extend the [ConceptsTest.qll](https://github.com/github/codeql/blob/main/python/ql/test/experimental/meta/ConceptsTest.qll) file, and add to the existing tests for [`fastapi`](https://github.com/github/codeql/tree/main/python/ql/test/library-tests/frameworks/fastapi) and [`starlette`](https://github.com/github/codeql/tree/main/python/ql/test/library-tests/frameworks/starlette).

**Comment by Kwstubbs** [comment]
@tausbn let me know if theres any other tests needed 🎈

**Review by tausbn — APPROVED** [review]
Thank you for the tests. This looks good to me! 🙂

**Comment by tausbn** [comment]
Oh, we should run a performance test before merging. I'll kick that off.

**Pull request merged**